packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -877,7 +890,10 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
     final bool enabled = widget.enabled ?? true;
     final Offset cursorOffset = Offset(_iOSHorizontalCursorOffsetPixels / MediaQuery.of(context)!.devicePixelRatio, 0);
     if (widget.maxLength != null && widget.maxLengthEnforced) {
-      formatters.add(LengthLimitingTextInputFormatter(widget.maxLength));
+      formatters.add(LengthLimitingTextInputFormatter(
+        widget.maxLength,
+        maxLengthEnforcement: widget.maxLengthEnforcement,
+      ));
```

Comment:
The logic here creates weird situations that will have to change later when the deprecated parameter is removed.

For example, what if maxLengthEnforced is false, but maxLengthEnforcement is `enforced`?  Currently, the length limit will not be enforced. However, after we remove the deprecated maxLengthEnforced parameter, the behavior will change and the limit will be enforced.

To avoid this, I think we should use some logic to get the effective maxLengthEnforcement based on both parameters.  Something like this:

```dart
MaxLengthEnforcement effectiveMaxLengthEnforcement;
// If maxLengthEnforced is the default, then just use maxLengthEnforcement.
if (widget.maxLengthEnforced) {
  effectiveMaxLengthEnforcement = widget.maxLengthEnforcement;
// Otherwise, maxLengthEnforced has been explicitly set to false, so use warningOnly.
} else {
  effectiveMaxLengthEnforcement = MaxLengthEnforcement.warningOnly;
}
```

You could do that in one line, but I'm just showing what I mean.  With that, once we deprecate maxLengthEnforced, then we can get rid of effectivemaxLengthEnforcement and just use maxLengthEnforcement directly with no behavior change.